src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -1926,6 +1928,504 @@ partial class C : I
+            }
+        }
+
+        [Theory, CombinatorialData, WorkItem("https://github.com/dotnet/roslyn/issues/77346")]
+        public void InInterface_Sealed(
+            [CombinatorialValues(LanguageVersion.Preview, LanguageVersionFacts.CSharpNext)] LanguageVersion langVersion)
+        {
+            var source = """
+                partial interface I
+                {
+                    public sealed partial int M();
+                    public sealed partial int M() => 0;
+                }
+                """;
+            var comp = CreateCompilation(source,
+                parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion),
+                targetFramework: TargetFramework.Net60).VerifyDiagnostics();
+            CompileAndVerify(comp,
+                sourceSymbolValidator: validate,
+                symbolValidator: validate,
+                verify: Verification.FailsPEVerify).VerifyDiagnostics();
+
+            static void validate(ModuleSymbol module)
+            {
+                var m = module.GlobalNamespace.GetMember<MethodSymbol>("I.M");
+                validateMethod(m);
+
+                if (module is SourceModuleSymbol)
+                {
+                    validateMethod((MethodSymbol)m.GetPartialImplementationPart()!);
+                }
+            }
+
+            static void validateMethod(MethodSymbol m)
+            {
+                Assert.False(m.IsAbstract);
+                Assert.False(m.IsVirtual);
+                Assert.False(m.IsMetadataVirtual());
+                Assert.False(m.IsMetadataNewSlot());
+                Assert.False(m.IsSealed);
+                Assert.False(m.IsStatic);
+                Assert.False(m.IsExtern);
+                Assert.False(m.IsOverride);
+                Assert.Equal(Accessibility.Public, m.DeclaredAccessibility);
+                Assert.True(m.ContainingModule is not SourceModuleSymbol || m.IsPartialMember());
+            }
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/77346")]
+        public void InInterface_Sealed_CSharp13()
+        {
+            var source = """
+                partial interface I
+                {
+                    public sealed partial int M();
+                    public sealed partial int M() => 0;
+                }
+                """;
+            var comp = CreateCompilation(source,
+                parseOptions: TestOptions.Regular13,
+                targetFramework: TargetFramework.Net60)
+                .VerifyDiagnostics(
+                    // (3,31): error CS0238: 'I.M()' cannot be sealed because it is not an override
+                    //     public sealed partial int M();
+                    Diagnostic(ErrorCode.ERR_SealedNonOverride, "M").WithArguments("I.M()").WithLocation(3, 31),
+                    // (4,31): error CS0238: 'I.M()' cannot be sealed because it is not an override
```

Comment:
>// (4,31): error CS0238: 'I.M()' cannot be sealed because it is not an override [](http://example.com/codeflow?start=20&length=79)

This look inconsistent with behavior for a regular method
 ```
public sealed int M() => 0;
```

Reply:
But it is consistent with behavior in previous version